array_methods_lab.js

```diff
@@ -0,0 +1,255 @@
+// Output: false
+
+
+array4 = [1,1,1,1,1]
+
+let newArr4 = array4.every((el) => {
+   return el === array4[0]
+})
+console.log(newArr4);
+
+console.log(" ");
+console.log("Question 5");
+// Write a function that returns the sum of every element in an array.
+//
+// Input: [1,2,3,4,5]
+// Output: 15
+
+let array5 = [1,2,3,4,5]
+
+let newArr5 = array5.reduce((acc, el) => {
+    return acc + el
+},0)
+
+console.log(newArr5);
+
+console.log(" ");
+console.log("Question 6");
+// Write a function that finds largest number in an array.
+//
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+
+
+
+let array6 = [1,2,6,3,4,5]
+
+let newArr6 = array6.reduce((acc, el) => {
+  if (acc > el){
+    return (acc);
+  }else{
+    return (el);
+  }
+})
+
+console.log(newArr6);
+
+console.log(" ");
+console.log("Question 7");
+// Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+//
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+//
+
+let array7 = [1,0,0,2,0,3,0,0,0]
+
+let newArr7 = array7.reduce((acc, el) => {
+      if(el === 0){
+          acc++
+          return acc
+      }else {
+        return acc;
+      }
+},0)
+
+console.log(newArr7);
+
+console.log(" ");
+console.log("Question 8");
+// Write a function that doubles every element in an array.
+//
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+
+let array8 = [1,2,3,4]
+
+array8.forEach((el) => {
+  console.log(el * 2);
+})
+
+console.log(" ");
+console.log("Question 9");
+// Write a function that finds the sum of every element in an array doubled
+//
+// Input: [1,2,3,4]
+// Output: 20
+
+
+let array9 = [1,2,3,4]
+
+let newArr9 = array9.reduce((acc, el) => {
+          return acc + el * 2
+}, 0)
+
+
+console.log(newArr9);
+
+
+// console.log("Question 10");
+// Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.  Return a new array of every element of arr that can be evenly
+// divided by divisor.
+//
+// Input: 5, [4,5,6,7,8,9,10,11,12,13,14,15]
+// Output: [5,10,15]
+//
+// let array10 = [4,5,6,7,8,9,10,11,12,13,14,15]
+//
+//
+// let elementDivisibleBy = array10.reduce((divisor, arr) =>{
+//   if ()
+// },5)
+
+console.log(" ");
+console.log("Question 11");
+// Write a `numberTimesIdx` that uses `map` and multiples each number in the array by its index.
+//
+// Input: [6,7,8,9]
+// Output: [0,7,16,27]
+
+let array11 = [6,7,8,9]
+
+const numberTimesIdx = (arr) => {
+  let result = arr.map((el, i) => {
+      return el * i
+    })
+    return result
+}
+console.log(numberTimesIdx(array11));
+
+console.log(" ");
+console.log("Question 12");
+// Write a function that returns whether or not every value in an array is a positive even number that doesn't end in 0.
+//
+// Input: [2,4,6,8,12]
+// Output: true
+//
+// Input: [2,4,6,8,10]
+// Output: false
+
+
+let array12 = [2,4,6,8,12]
+
+let result12 = array12.every((el) => {
+  if(el > 0 && el % 10 !== 0){
+    return true
+  }else{
+    return false
+  }
+})
+
```

Comment:
Great . Remember every returns a boolean. So if you return the condition it'll be fine.